### ISSUE TITLE
Separate Meals and Recipes

### ISSUE THREAD

**Issue by F3PiX** [F3PiX/HansCookBook#51]
Most recipes are actually meals. A meal has (many) recipes with subchoices, for instance: which recipe for pasta with chicken, which tomatosauce, which salad or vegetables.
Maybe some subchoices are accompaniments, like pico de gallo, kroepoek. 
There are also 'best friends' food pairing techniques.

- [ ] User.profile should give the opportunity to select what kind of meals to include (first proteintypes, then refine) and to add some
- [ ] Recipes should help with special technique tips
- [ ] A User has many meals: in first pass, divide current recipes in MEALS and RECIPES
- [ ] A MEAL also has a MEAL-TYPE: quick and easy, take away, festive, improvise.

- [ ] Should a MEAL have a season? Some do
- [ ] Allow for a bitmask to combine filters
- [ ] User.profile has a cooking schedule, also shopping days
- [ ] 

Requirements;
- [ ]  User (not profile) has many `:meals`
- [ ] Meals are the same for everyone, filtered as preferences in User.profile 
- [ ] Select a meal like recipes are chosen currently
  - [ ] Change "recipe" to 'meal'
  - [ ] Everything should work the same
  - [ ] Create new table 'recipes', and migrate 'recipe details from 'meal' back into 'recipe'
  - [ ] Remove the recipe fields from MEAL, 
 
 MEAL has many recipes
 User chooses meal, -> then recipe . cook button on recipe. Update meal as 'chosen', update recipe as uncookable for a period. 
Meal: 
1) pasta with chicken + redshit              
2) pasta with chicken + white shit
3) pasta salad with left over chicken. 
4) 
Recipes: 
1.1)  
2.1) with kaassaus
2.2) with champignons

**Comment by F3PiX** [comment]
On second thought... I don't think this is the way to go.  It doesn't seem  to improve the user experience to choose a meal first. And what needs to be solved, might be taken care of with a :meal_category kind of thing on the recipe. 
Then adding recipe has_many variants, and has_many companions, combined with the uncookable status.

TL;DR Don't do this.
